File: src/tests/stuff/moretest/palindrome.go
package main

import "fmt"

func main() {
	fmt.Println("palindrome")
	s := "amanaplanacanalpanama"
	i := 0
	j := len(s) - 1
	isPal := true
	for i < j {
		if s[i] != s[j] {
			isPal = false
			break
		}
		i++
		j--
	}
	fmt.Println(isPal)
}

File: src/tests/stuff/moretest/palindrome.rs
fn main() {
    println!("palindrome");
    let s = "amanaplanacanalpanama";
    let mut i = 0;
    let mut j = s.len() - 1;
    let mut is_pal = true;
    let bytes = s.as_bytes();
    while i < j { if bytes[i] != bytes[j] { is_pal = false; break; } i += 1; j -= 1; }
    println!("{}", if is_pal { "True" } else { "False" });
}
